packages/twenty-front/src/modules/object-record/hooks/useRecordOneToManyFieldAttachTargetRecord.ts

```diff
@@ -0,0 +1,123 @@
+  const { fieldDefinition } = useContext(FieldContext);
+
+  const { objectMetadataItems } = useObjectMetadataItems();
+  const { objectPermissionsByObjectMetadataId } = useObjectPermissions();
+  const { updateOneRecord } = useUpdateOneRecordV2();
+
+  const recordOneToManyFieldAttachTargetRecord = async ({
+    sourceObjectNameSingular,
+    targetObjectNameSingular,
+    targetGQLFieldName,
+    sourceRecordId,
+    targetRecordId,
+  }: {
+    sourceObjectNameSingular: string;
+    targetObjectNameSingular: string;
+    targetGQLFieldName: string;
+    sourceRecordId: string;
+    targetRecordId: string;
+  }) => {
+    const targetObjectMetadataItem = objectMetadataItems.find(
+      (objectMetadataItem) =>
+        objectMetadataItem.nameSingular === targetObjectNameSingular,
+    );
+
+    const sourceObjectMetadataItem = objectMetadataItems.find(
+      (objectMetadataItem) =>
+        objectMetadataItem.nameSingular === sourceObjectNameSingular,
+    );
+
+    if (!targetObjectMetadataItem) {
+      throw new CustomError(
+        `Could not find related object metadata for ${targetObjectNameSingular}`,
+        'RELATED_OBJECT_METADATA_NOT_FOUND',
+      );
+    }
+
+    if (!sourceObjectMetadataItem) {
+      throw new CustomError(
+        `Could not find source object metadata for ${sourceObjectNameSingular}`,
+        'SOURCE_OBJECT_METADATA_NOT_FOUND',
+      );
+    }
+
+    const cachedTargetRecord = getRecordFromCache({
+      objectMetadataItem: targetObjectMetadataItem,
+      recordId: targetRecordId,
+      cache: apolloCoreClient.cache,
+      objectMetadataItems,
+      objectPermissionsByObjectMetadataId,
+    });
+
+    if (!cachedTargetRecord) {
+      throw new Error('Could not find cached related record');
+    }
+
+    const previousRecordId = cachedTargetRecord?.[`${targetGQLFieldName}Id`];
+
+    if (isDefined(previousRecordId)) {
+      const previousRecord = getRecordFromCache({
+        objectMetadataItem: sourceObjectMetadataItem,
+        recordId: previousRecordId,
+        cache: apolloCoreClient.cache,
+        objectMetadataItems,
+        objectPermissionsByObjectMetadataId,
+      });
+
+      const previousRecordWithRelation = {
+        ...cachedTargetRecord,
+        [targetGQLFieldName]: previousRecord,
+      };
+
+      const gqlFields = computeDepthOneRecordGqlFieldsFromRecord({
+        objectMetadataItem: targetObjectMetadataItem,
+        record: previousRecordWithRelation,
+      });
+
+      updateRecordFromCache({
+        objectMetadataItems,
+        objectMetadataItem: targetObjectMetadataItem,
+        cache: apolloCoreClient.cache,
+        record: {
+          ...cachedTargetRecord,
+          [targetGQLFieldName]: previousRecord,
+        },
+        recordGqlFields: gqlFields,
+        objectPermissionsByObjectMetadataId,
+      });
+    }
+
+    assertFieldMetadata(
+      FieldMetadataType.RELATION,
+      isFieldRelation,
+      fieldDefinition,
```

Comment:
i am having a bit of trouble understanding how this can work for both morph and relation. fielddefintion inherited from the fieldContext should fail when it's a morph